packages/core/review-workflows/server/src/services/workflows.ts

```diff
@@ -128,11 +140,36 @@ export default ({ strapi }: { strapi: Core.Strapi }) => {
 
           updatedStageIds = await stageService
             .replaceStages(workflow.stages, opts.data.stages, workflow.contentTypes)
-            .then((stages: any) => stages.map((stage: any) => stage.id));
+            .then((stages: any) => {
+              updatedStages = stages;
+              return stages.map((stage: any) => stage.id);
+            });
```

Comment:
I find the mix of functional and non-functional code a bit weird here (since you mutate updated stages). I would recommend doing it in two steps instead:

- `const updatedStages = await stageService.replaceStages(...)`
- `const updatedStagesIds = updatedStages.map(stage => stage.id)`